karate-core/src/main/java/com/intuit/karate/StepDefs.java

```diff
@@ -364,6 +371,18 @@ private void unprepare() {
         multiPart = null;
         request = null;
     }
+    private Object getRawResponse() {
+        //If a byte array contains a negative-signed byte,
+        //then the string conversion will corrupt it. In
+        //that case just return the byte array stream
+        byte[] rawBytes = response.readEntity(byte[].class);
```

Comment:
When reading response to byte array, we should define charset(like UTF-8), otherwise strings in other than default format will not be read properly.

Reply:
@iamshahjahan you are looking at outdated code. please stop